sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -243,40 +243,26 @@ case class Cbrt(child: Expression) extends UnaryMathExpression(math.cbrt, "CBRT"
   examples = """
     Examples:
       > SELECT _FUNC_(-0.1);
-       0
+       0.0
       > SELECT _FUNC_(5);
```

Comment:
For an integral type with scale parameter 0, we will get an integral type as the output. We cover this case in the unit test `checkEvaluation(Ceil(5, 0), 5)`. 

We could keep the output type to bigInt if the scale parameter is 0 for `floor`/`ceil` but the behavior will be different than that of `round`/`bround`.

Reply:
@cloud-fan @gengliangwang  For the Float and Double type during ROUND_CEILING and ROUND_FLOOR with scale 0, shall we cast the output type to Long ? This will preserve the default behavior of `floor`/`ceil`. Please let me know your opinion.

something like this :
```
  def castOutputToLong: Boolean = Seq(ROUND_CEILING, ROUND_FLOOR).contains(mode) && _scale == 0

     ...
  def nullSafeEval(input1: Any): Any = {
    dataType match {
      ...

      case FloatType =>
        val f = input1.asInstanceOf[Float]
        val output = if (f.isNaN || f.isInfinite) {
          f
        } else {
          BigDecimal(f.toDouble).setScale(_scale, mode).toLong
        }
        if(castOutputToLong) output.toLong else output.toShort

      case DoubleType =>
        val d = input1.asInstanceOf[Double]
        val output = if (d.isNaN || d.isInfinite) {
          d
        } else {
          BigDecimal(d).setScale(_scale, mode).toLong
        }
        if(castOutputToLong) output.toLong else output.toShort
    }
  }
```